### PR TITLE
Make it possible to sell unsold privates during SR.

### PR DESCRIPTION
**Pull request by perwestling**
This is a preparation for 1824 which does not have an initial auction,
but instead the privates are sold together with shares during SR 1.

### CODE REVIEW

File: lib/engine/step/buy_sell_par_shares.rb [hunk 1]
```diff
@@ -211,6 +211,10 @@ def purchasable_companies(entity)
         @game.purchasable_companies(entity)
       end
 
+      def purchasable_unsold_companies
+        @game.companies.select { |c| c.owner.nil? }
```
Comment (tobymao) [inline lib/engine/step/buy_sell_par_shares.rb:215]: companies.reject{ |c| c.owner }

File: lib/engine/step/buy_sell_par_shares.rb [hunk 2]
```diff
@@ -239,15 +243,18 @@ def process_buy_company(action)
...
         company.owner = entity
-        owner.companies.delete(company)
+        owner&.companies&.delete(company)
 
         entity.companies << company
-        entity.spend(price, owner)
+        entity.spend(price, owner.nil? ? @game.bank : owner)
         @current_actions << action
-        @log << "-- #{entity.name} buys #{company.name} from #{owner.name} for #{@game.format_currency(price)}"
+        @log << "#{owner.nil? ? '' : '-- '}"\
```
Comment (tobymao) [inline lib/engine/step/buy_sell_par_shares.rb:254]: owner ? '--' ..

File: lib/engine/step/buy_sell_par_shares.rb [hunk 3]
```diff
@@ -239,15 +243,18 @@ def process_buy_company(action)
...
+                "#{entity.name} buys #{company.name} from "\
+                "#{owner.nil? ? 'the market' : owner.name} for "\
```
Comment (tobymao) [inline lib/engine/step/buy_sell_par_shares.rb:256]: same here inverse so you don't have to call explicit nil

File: lib/engine/step/buy_sell_par_shares.rb (hunk 1 above)
Comment (perwestling) [reply lib/engine/step/buy_sell_par_shares.rb]: Fixed

File: lib/engine/step/buy_sell_par_shares.rb (hunk 2 above)
Comment (perwestling) [reply lib/engine/step/buy_sell_par_shares.rb]: Fixed

File: lib/engine/step/buy_sell_par_shares.rb (hunk 3 above)
Comment (perwestling) [reply lib/engine/step/buy_sell_par_shares.rb]: Fixed

**Pull request merged**